### DISCUSSION TITLE
メンバーという概念を導入する？

### DISCUSSION THREAD

**Discussion by h-takeyeah** [su-its/ams-project#4] [Ideas]
# メンバーという概念
以前から言っていた話なのですが，予め研究室のメンバーを登録しておくというのはどうでしょうか．どうでしょうかというのは実現可能性と技術的な障壁，時間や優先順位を考えて開発しても大丈夫でしょうかという意味です．

## 研究室のメンバーを登録しておくことで得られる恩恵
- ログが名前付きで出せる
- 研究室のメンバー以外は記録しないという機能が使える

## 実装方法案
メンバーの学籍番号と名前を保存するテーブルを新規に作る．

**Comment by equal-l2** [comment]
たぶんその機能の追加はそんなに難しくはないと思います。
ただ簡単なわりに優先度の高い機能ではないので、4月以降に作っていいと思います。

---
以下、必要な追加実装

## DB
- 「学籍番号-名前」対応表（以下単に「対応表」）の追加

## バックエンド
- `POST /room`のハンドラでメンバー以外を弾く処理
- 対応表参照・変更用のAPI
- 認証？（誰でも触れるようにしていいならいらない）

## フロントエンド
- ログ表示の際に学籍番号から名前を解決する処理
- 対応表操作関係のUI

**Reply by equal-l2** [reply]
認証は「難しい」に入るかも

**Reply by h-takeyeah** [reply]
分かりました。納品には含めないということで、いったん保留にしておきます。そういえばこのDiscussionsという機能には保留マークとかないんでしょうか？pendingできると嬉しいのですが。

**Reply by equal-l2** [reply]
そういうのはないですね

**Reply by equal-l2** [reply]
IssueからDiscussionsへの変換はできるんですが、逆はできないみたいです
